src/OrchardCore.Modules/OrchardCore.Email/Controllers/AdminController.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            if ( !await _authorizationService.AuthorizeAsync( User, Permissions.ManageEmailSettings ) )
+            {
+                return Unauthorized();
+            }
+
+            var logger = new FakeLogger<SmtpService>();
+
+            if ( !ModelState.IsValid )
+            {
+                logger.LogError( "Invalid settings" );
+                return Json( new { error = logger.Message } );
+            }
+
+            var smtp = new SmtpService( Options.Create<SmtpSettings>( testSettings ), logger );
+
+            try
+            {
+                await smtp.SendAsync( new EmailMessage
+                {
+                    From = testSettings.From,
+                    Recipients = testSettings.To,
+                    Cc = testSettings.Cc,
+                    Bcc = testSettings.Bcc,
+                    ReplyTo = testSettings.ReplyTo,
+                    Subject = testSettings.Subject,
+                    Body = testSettings.Body,
+                } );
+
+
+                if ( !String.IsNullOrEmpty( logger.Message ) )
+                {
+                    return Json( new { error = logger.Message } );
+                }
+
+                return Json( new { status = T["Message sent."].Value } );
+            }
+            catch ( Exception e )
+            {
+                return Json( new { error = e.Message } );
+            }
+
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> SmtpSavedSettingsTest(TestSmtpOptions testSettings, [FromServices]ISmtpService smtp)
+        {
+            if ( !await _authorizationService.AuthorizeAsync( User, Permissions.ManageEmailSettings ) )
+            {
+                return Unauthorized();
+            }
+
+            var logger = new FakeLogger<SmtpService>();
+
+            if ( !ModelState.IsValid )
+            {
+                logger.LogError( "Invalid settings" );
+                return Json( new { error = logger.Message } );
+            }
+
+            
+            try
+            {
+                await smtp.SendAsync( new EmailMessage
+                {
+                    From = testSettings.From,
+                    Recipients = testSettings.To,
+                    Cc = testSettings.Cc,
+                    Bcc = testSettings.Bcc,
+                    ReplyTo = testSettings.ReplyTo,
+                    Subject = testSettings.Subject,
+                    Body = testSettings.Body,
+                } );
+
+
+                if ( !String.IsNullOrEmpty( logger.Message ) )
+                {
+                    return Json( new { error = logger.Message } );
+                }
+
+                return Json( new { status = T["Message sent."].Value } );
+            }
+            catch ( Exception e )
+            {
+                return Json( new { error = e.Message } );
+            }
+        }
+
+
+        private class FakeLogger<T> : ILogger<T>
```

Comment:
Fake Logger?

Reply:
Yes, copied from O1.  It is used for reporting back any issue to UI.

Reply:
We need to fix it. If errors need to be reported, let's do it like for user validation, with LocalizedStrings

Reply:
@sebastienros I am looking at it right now.  Actually this is for the ajax call to test the smtp settings.  So are you suggesting I should return LocalizedStrings in the returned json?

Reply:
Don't do anything on it right now, once I look more closely I will let you know what to do

Reply:
Ok.

Reply:
I don't think we need an ajax call. The test button should just issue a POST to a custom action that will be like the Save button POST, but without saving the values (just to keep to forms with the edited values) and also trying to send an email. The errors are then displayed like validation errors using `AddModelError`.